src/gc/collector.cpp

```diff
@@ -273,8 +371,8 @@ void markPhase() {
     VALGRIND_DISABLE_ERROR_REPORTING;
 #endif
 
-    TraceStack stack(roots);
-    GCVisitor visitor(&stack);
+    shared_ptr<MarkStack> stack = make_shared<MarkStack>(roots);
```

Comment:
missing `std::`'s here.  we should probably not have the `using namespace std` at the beginning of the file.